Documentation/API.md

```diff
@@ -641,36 +644,41 @@ extend CanBeExtended {
+
 * **Extension objects** are opaque objects that define the extension itself,
   including storage and serialization details.  Because proto allows extension
-  names to be reused in different scopes, these objects appear in the scope corresponding to the
-  context where the proto extension was defined, which generally does not
-  correspond to that of the message being extended.  In the above example, the
-  extension object would be `Extensions_extensionField` at the file scope.
-* **Extension maps** are collections of extension objects indexed by the target
-  message and field number.  An extension map is
-  generated by the code generator and included as a static global variable in
-  every Swift source file generated from a proto file that defines extensions.
-  Extension maps are provided to the deserialization APIs to define what
-  extension fields the decoder should recognize.  In the above example, the
-  generated extension map would be `Sample_Extensions`.
+  names to be reused in different scopes, these objects appear in the scope
+  corresponding to the context where the proto extension was defined (file
+  level or within the message that wrapped the `extend` directive); generally
+  it does not correspond to that of the message being extended.  In the above
+  example, the extension object would be `Extensions_extensionField` at the
+  file scope. Most common Swift code accessing Extensions won't have to access
+  these directly.
+
+* **Extension properties** use Swift's `extension` capability to add properties
+  to the message that is being extended.  In most cases, you can simply use
+  the extension properties without understanding any of the other extension
+  machinery.  The above example creates a Swift extension of `CanBeExtended`
+  that defines a new property `extensionField` of type `Int32`.
 
-```swift
-/// ExtensionMap is a protocol; you can define your own optimized
-/// form for special applications.  The standard implementation in the
-/// library is `SimpleExtensionMap`.
-public protocol ExtensionMap {
-    public subscript(messageType: Message.Type, protoFieldNumber: Int) -> MessageExtensionBase? {get}
-    public func fieldNumberForProto(messageType: Message.Type, protoFieldName: String) -> Int?
-}
-```
+* **Extension maps** are collections of extension objects indexed by the target
+  message and field number.  An extension map is generated for every file that
+  defined proto extensions and included as a static global variable.  It is
+  named based on the proto package, filename, and then ends in `_Extensions`,
+  so the above file would be `Sample_Extensions`. These maps are then used
+  the the `Message` apis for parsing/merging binary data to provide the
+  additional data to parse out the extension fields; if a mapping isn't
+  found, the extension field ends up in the `unknownFields` on the message.
+
+  If you need handle extensions defined in multiple files, you can build up
```

Comment:
"If you need to handle..."  ---- omitted word "to"

Reply:
done